docs/source/user_guide/installation/install-from-docker.rst

```diff
@@ -0,0 +1,221 @@
+
+.. tab:: ODES Solver
+
+      .. code:: bash
+
+            docker pull pybamm/pybamm:odes
+
+.. tab:: JAX Solver
+
+      .. code:: bash
+
+            docker pull pybamm/pybamm:jax
+
+.. tab:: IDAKLU Solver
+
+      .. code:: bash
+
+            docker pull pybamm/pybamm:idaklu
+
+.. tab:: All Solver
+
+      .. code:: bash
+
+            docker pull pybamm/pybamm:all
+
+Running the Docker container
+----------------------------
+
+Once you have pulled the Docker image, you can run a Docker container with the PyBaMM environment:
+
+1. In your terminal, use the following command to start a Docker container from the pulled image:
+
+.. tab:: Basic
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:latest
+
+.. tab:: ODES Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:odes
+
+.. tab:: JAX Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:jax
+
+.. tab:: IDAKLU Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:idaklu
+
+.. tab:: All Solver
+
+      .. code:: bash
+
+            docker run -it pybamm/pybamm:all
+
+2. You will now be inside the Docker container's shell. You can use PyBaMM and its dependencies as if you were in a virtual environment.
+
+3. You can execute PyBaMM-related commands, run tests develop & contribute from the container.
+
+Exiting the Docker container
+----------------------------
+
+To exit the Docker container's shell, you can simply type:
+
+.. code-block:: bash
+
+      exit
+
+This will return you to your host machine's terminal.
+
+Building Docker image locally from source
+-----------------------------------------
+
+If you want to build the PyBaMM Docker image locally from the PyBaMM source code, follow these steps:
+
+1. Clone the PyBaMM GitHub repository to your local machine if you haven't already:
+
+.. code-block:: bash
+
+      git clone https://github.com/pybamm-team/PyBaMM.git
+
+2. Change into the PyBaMM directory:
+
+.. code-block:: bash
+
+      cd PyBaMM
+
+3. Build the Docker image using the following command:
+
+.. code-block:: bash
+
+      docker build -t pybamm -f scripts/Dockerfile .
+
+4. Once the image is built, you can run a Docker container using:
+
+.. code-block:: bash
+
+      docker run -it pybamm
+
+5. Activate PyBaMM development environment inside docker container using:
+
+.. code-block:: bash
+
+      conda activate pybamm
+
+Building Docker images with optional args
+-----------------------------------------
+
+When building the PyBaMM Docker images locally, you have the option to include specific solvers by using optional arguments. These solvers include:
+
+- ``IDAKLU``: For IDA solver provided by the SUNDIALS plus KLU.
+- ``ODES``: For scikits.odes solver for ODE & DAE problems.
+- ``JAX``: For Jax solver.
+
+To build the Docker images with optional arguments, you can follow these steps for each solver:
+
+Build Docker image with IDAKLU solver
```

Comment:
Could you arrange these optional solver texts into tabs? Just like the ones above!

Reply:
Done